### PR TITLE
docs(react): update Tailwind CSS guide for v3

### LINKED ISSUE #8218: React: Tailwind CSS 3 doesn't work
## Current Behavior

If you try to add TailwindCSS to a React project, the tailwind styles are never generated when serving or buildings.

## Expected Behavior

- Tailwind styles are applied

## Steps to Reproduce

- Create a new workspace
- Add the react plugin
- Create a new react application
- Follow the instructions at: https://nx.dev/l/r/guides/using-tailwind-css-in-react
  - Note, using the new `content` key instead of `purge`
- Serve the app

### Failure Logs

- No failure logs

### Environment

```
  Node : 14.18.1
  OS   : darwin x64
  yarn : 1.22.17
  
  nx : 13.3.6
  @nrwl/angular : undefined
  @nrwl/cli : 13.3.6
  @nrwl/cypress : 13.3.9
  @nrwl/devkit : 13.3.9
  @nrwl/eslint-plugin-nx : 13.3.9
  @nrwl/express : undefined
  @nrwl/jest : 13.3.9
  @nrwl/linter : 13.3.9
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : 13.3.9
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.3.6
  @nrwl/web : 13.3.9
  @nrwl/workspace : 13.3.6
  @nrwl/storybook : 13.3.9
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 6.6.7
  ---------------------------------------
  Community plugins:
```

### LINKED ISSUE #8799: Outdated Tailwind CSS config file
Please update tailwind features here: https://nx.dev/guides/using-tailwind-css-in-react#using-tailwind-css-in-react

### PR DESCRIPTION
**Pull request by puku0x**
## Current Behavior

https://nx.dev/l/r/guides/using-tailwind-css-in-react is outdated, for example, `purge` has been renamed `content`.

## Expected Behavior

The guide follows the latest Tailwind CSS docs.
https://tailwindcss.com/docs/installation

## Related Issue(s)

Closes #8218
Closes #8799
Refs #8355
Refs #8594

### CODE REVIEW

File: docs/react/guides/using-tailwind-css-in-react.md [hunk 1]
```diff
@@ -65,7 +70,7 @@ _NOTE:_ To ensure proper purging for custom configurations, be sure that the `NO
 Next, import tailwind styles to the application's base `styles.css` or `styles.scss` file. This can be done by adding the following lines:
 
 ```css
-@tailwind components;
 @tailwind base;
+@tailwind components;
 @tailwind utilities;
```
Comment (puku0x) [inline docs/react/guides/using-tailwind-css-in-react.md:75]: image

**Comment by smacintyre** [comment]
A couple notes:

- The text still talks about the deprecated `purge` key
- The dark mode configuration should be removed: https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration
- The variant config should be removed: https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration

File: docs/react/guides/using-tailwind-css-in-react.md [hunk 2]
```diff
@@ -30,30 +30,31 @@ Next, adjust the `postcss.config.js` as follows:
...
+In a typical `tailwind.config.js` file, the `content` property of the tailwind config would be an array that includes all files that could mention tailwind class names (you can find more details on tailwind's [official documentation](https://tailwindcss.com/docs/content-configuration)).
 
 Nx has a utility function for determining the glob representation of all files the application depends on (based on the Nx Dependency Graph), which should be used when setting this purge property. This eliminates additional manual maintenance as your workspace progresses.
 
 ```js
 const { createGlobPatternsForDependencies } = require('@nrwl/react/tailwind');
 
 module.exports = {
-  purge: createGlobPatternsForDependencies(__dirname),
-  darkMode: false, // or 'media' or 'class'
```
Comment (puku0x) [inline docs/react/guides/using-tailwind-css-in-react.md:50]: https://github.com/nrwl/nx/issues/8218#issuecomment-997189329
> The dark mode configuration should be removed: https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration

File: docs/react/guides/using-tailwind-css-in-react.md [hunk 3]
```diff
@@ -30,30 +30,31 @@ Next, adjust the `postcss.config.js` as follows:
...
+  content: [
+    'apps/{your app here}/src/**/*.{js,jsx,ts,tsx}',
+    ...createGlobPatternsForDependencies(__dirname),
+  ],
   theme: {
     extend: {},
   },
-  variants: {
-    extend: {},
-  },
```
Comment (puku0x) [inline docs/react/guides/using-tailwind-css-in-react.md:56]: https://github.com/nrwl/nx/issues/8218#issuecomment-997189329
> The variant config should be removed: https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration

**Pull request closed without merging**